soql-analyzer/src/main/scala/com/socrata/soql/AnalysisDeserializer.scala

```diff
@@ -207,38 +254,57 @@ class AnalysisDeserializer[C, T](columnDeserializer: String => C, typeDeserializ
         dictionary.strings(in.readUInt32())
       }
 
+    def readAnaAndFrom(): (SoQLAnalysis[C, T], Option[String]) = {
+      lazy val fromOptTrigger = if (isV4) maybeRead(in.readString) else None
+
+      val ig = readIsGrouped()
+      val d = readDistinct()
+      val s = readSelection()
+      val j = {
+        fromOptTrigger; // trigger eval, read "from" string if v4
+        readJoins()
+      }
+      val w = readWhere()
+      val gb = bcSeq { readGroupBy() }
+      val h = readHaving()
+      val ob = bcSeq { readOrderBy() }
+      val l = readLimit()
+      val o = readOffset()
+      val search = readSearch()
+
+      val ana = SoQLAnalysis(ig, d, s, j, w, gb, h, ob, l, o, search)
+
+      (ana, fromOptTrigger)
+    }
 
     def readAnalysis(): SoQLAnalysis[C, T] = {
-      SoQLAnalysis(
-        readIsGrouped(),
-        readDistinct(),
-        readSelection(),
-        readJoins(),
-        readWhere(),
-        readGroupBy(),
-        readHaving(),
-        readOrderBy(),
-        readLimit(),
-        readOffset(),
-        readSearch())
+      readAnaAndFrom()._1
     }
 
     def read(): NonEmptySeq[SoQLAnalysis[C, T]] = {
       readNonEmptySeq { readAnalysis() }
     }
+
+    def readWithFrom(): NonEmptySeq[(SoQLAnalysis[C, T], Option[String])] = {
+      readNonEmptySeq { readAnaAndFrom() }
+    }
   }
 
   def apply(in: InputStream): NonEmptySeq[SoQLAnalysis[C, T]] = {
     val cis = CodedInputStream.newInstance(in)
     cis.readInt32() match {
-      case 0 =>
+      case v if v == TestVersionV4 || v == TestVersionV5 => // single select
         val dictionary = DeserializationDictionaryImpl.fromInput(cis)
-        val deserializer = new Deserializer(cis, dictionary, 0)
+        val deserializer = new Deserializer(cis, dictionary, v)
         NonEmptySeq(deserializer.readAnalysis(), Seq.empty)
       case v if v >= 3 && v <= 4 =>
```

Comment:
Could this be changed to use `isV4` to make it easier to strip out the code when the rollout is finished? Does v3 exist anywhere anymore?

Reply:
gonna blast all this code immediately following release of 2.11.0 anyway, so i'm tempted just to leave it to be extra safe and then get rid of it